engines/rancher_on_aks/app/models/rancher_on_aks/deployment.rb

```diff
@@ -64,10 +75,13 @@ def deploy()
           name: "#{@prefix}_cluster",
           resource_group: @resource_group,
           k8s_version: Rails.configuration.x.rancher_on_aks.k8s_version,
-          vm_size: "Standard_D2_v3",
-          node_resource_group: "#{@prefix}_nodes"
+          vm_size: @cluster_size.instance_type,
```

Comment:
should we add some error handling in (unlikely?) case `Standard_D2as_v5` is not available any more ?